### PR TITLE
feat: common API calls

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

The PR contains some common extracted code 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,8 +57,9 @@
         "lint-staged": "^12.5.0",
         "glob": "^10.3.3",
         "moment": "^2.29.4",
-        "monaco-editor": "0.39.0",
-        "monaco-yaml": "4.0.4",
+        "monaco-editor": "0.44.0",
+        "monaco-yaml": "5.1.1",
+        "react-monaco-editor": "^0.55.0",
```
Comment (eshankvaish) [inline package.json:62]: Should we add this to peer deps as well to ensure only single copy is created?

File: src/Common/CodeEditor/CodeEditor.tsx [hunk 2]
```diff
@@ -1,108 +1,44 @@
-import React, { useEffect, useCallback, useReducer, useRef } from 'react'
+import React, { useEffect, useReducer, useRef } from 'react'
 import MonacoEditor, { MonacoDiffEditor } from 'react-monaco-editor'
-// import { useJsonYaml, Select, RadioGroup, useWindowSize, copyToClipboard } from '../common'
-// import { Progressing } from '@devtron-labs/devtron-fe-common-lib'
 import YAML from 'yaml'
+import ReactGA from 'react-ga4'
```
Comment (eshankvaish) [inline src/Common/CodeEditor/CodeEditor.tsx:4]: Suggestion: we can remove the same from eslintignore

File: src/Common/CodeEditor/CodeEditor.tsx [hunk 3]
```diff
@@ -113,275 +49,245 @@ function useCodeEditorContext() {
...
+    noParsing = false,
+    defaultValue = '',
+    children,
+    tabSize = 2,
+    lineDecorationsWidth = 0,
+    height = 450,
+    inline = false,
+    shebang = '',
+    minHeight,
+    maxHeight,
```
Comment (eshankvaish) [inline src/Common/CodeEditor/CodeEditor.tsx:64]: potentially unused variable

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Common/CodeEditor/CodeEditor.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/Common/CodeEditor/CodeEditor.tsx:64]: Fixed

File: src/Common/CodeEditor/CodeEditor.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/Common/CodeEditor/CodeEditor.tsx:4]: Done

File: src/Common/CodeEditor/CodeEditor.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        case 'setCode':
+            return { ...state, code: action.value }
+        case 'setHeight':
+            return { ...state, height: action.value.toString() }
+        default:
+            return state
+    }
+}
+
+export const initialState = (mode, theme, value, diffView, noParsing): CodeEditorState => ({
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/Common/CodeEditor/CodeEditor.reducer.ts]: Done

File: package.json (hunk 1 above)
Comment (vivek-devtron) [reply package.json]: Fixed for now. although we will eventually remove these from dashboard

**Review by eshankvaish — APPROVED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by eshankvaish — DISMISSED** [review]

**Review by eshankvaish — APPROVED** [review]

**Pull request merged**